api/lang/diagnostic.go

```diff
@@ -21,7 +21,15 @@ func createDiagnostics(uri string) []lsp.Diagnostic {
 
 func createValidationDiagnostics(errors []validation.StepValidationError, uri string) (diagnostics []lsp.Diagnostic) {
 	for _, err := range errors {
-		diagnostics = append(diagnostics, createDiagnostic(err.Message(), err.Step().LineNo-1, 1, uri))
+		diagnostics = append(diagnostics, lsp.Diagnostic{
```

Comment:
Why can't we use `createDiagnostic` function here?

Reply:
We are adding one extra field (`Code`) in diagnostic for validation error. `createDiagonastic` doesn't  return a `pointer` (because we don't need one) so we cant change that after creating.